### ISSUE TITLE
Shipping Zones: Look into missing locations

### ISSUE THREAD

**Issue by psealock** [woothemes/woocommerce#40643]
**Prerequisites (mark completed items with an [x]):**
- [x] I have checked that my issue type is not listed here https://github.com/woocommerce/woocommerce/issues/new/choose
- [x] My issue is not a security issue, support request, bug report, enhancement or feature request (Please use the link above if it is).

**Issue Description:**

There have been reports of users unable to set locations when adding Shipping Zones.

See https://airtable.com/appra4zMRgiGscWB7/tblqKcO8HXWoSFKUR/viwVKXWZNUe7840Lb/recBOhxl2CFVmOIGl?blocks=hide

Is there a bug or are some locations missing? If so, whats the best way to handle it?

**Comment by psealock** [comment]
WooCommerce uses [CLDR](https://github.com/unicode-org/cldr/blob/release-42/common/subdivisions/en.xml) as a standard for gathering continents, countries, and states/regions/provinces. The standard is available on Github and an API solution to have continually updated locale data would greatly reduce merchant and customer pain while mitigating unforeseen problems with manually updating the hard coded files. Creating an API resource will require dedicated development work and should be prioritised in the context WooCommerce goals.

**TL;DR** - A CLDR API resource would be amazing, but there is no drop in solution - development time will be required. In the meantime, [manual updates](https://github.com/woocommerce/woocommerce/pull/41769) required.

### Plugin Guidelines

This sort of thing will be allowed by from third party CDN.

>Using third party services to manage regularly updated lists of data, when not explicitly permitted in the service’s terms of use

~https://developer.wordpress.org/plugins/wordpress-org/detailed-plugin-guidelines/#8-plugins-may-not-send-executable-code-via-third-party-systems

If WooCommerce discloses usage, it appears acceptable.

### Prior Art

There is already a plugin - https://github.com/Automattic/wp-cldr
We already do some data extraction and manipulation from source data - https://github.com/woocommerce/woocommerce/pull/40101
Example manual update - https://github.com/woocommerce/woocommerce/pull/41769

### Existing CDN?

The closest I can find is https://cdnjs.com/libraries/cldr-json/43.1.0, which has the raw data in JSON format. In order to extract just the data we need, some manipulation will be involved so I don't think this CDN could be used directly. There is also source on Github, but again some manipulation required prevents us from pinging the raw files directly.
